care/facility/api/serializers/shifting.py

```diff
@@ -182,16 +244,25 @@ def create(self, validated_data):
 
         validated_data["is_kasp"] = False
 
-        orgin_facility_external_id = validated_data.pop("orgin_facility")["external_id"]
+        # orgin_facility_external_id = validated_data.pop("orgin_facility")["external_id"]
         # validated_data["orgin_facility_id"] = Facility.objects.get(external_id=orgin_facility_external_id).id
 
-        shifting_approving_facility_external_id = validated_data.pop("shifting_approving_facility")["external_id"]
+        shifting_approving_facility_external_id = validated_data.pop(
+            "shifting_approving_facility"
+        )["external_id"]
         validated_data["shifting_approving_facility_id"] = Facility.objects.get(
             external_id=shifting_approving_facility_external_id
         ).id
 
-        if "assigned_facility" in validated_data:
-            assigned_facility_external_id = validated_data.pop("assigned_facility")["external_id"]
+        if validated_data.get("assigned_facility_external"):
+            validated_data["assigned_facility"] = None
+        elif validated_data.get("assigned_facility"):
+            validated_data["assigned_facility_external"] = None
+
+        if validated_data.get("assigned_facility"):
```

Comment:
This condition is just above it as well right?

Reply:
The above one is for the `update` method, this one is in the `create` method.

Reply:
Line 262 and 259 are the same condition right ?

Reply:
Yep those two are the same.. I had separated those code blocks as they served different purposes. The purpose of the initial if else block was to clear out the "other" facility source, so that both external and care facilities could not be set together.

The second `if` block was to set the value of `assigned_facility_id` (old code, already present).

I can proceed with merging those two conditions. So, it'll look like: (Basically merge those two `if` blocks)

```
        if validated_data.get("assigned_facility_external"):
            validated_data["assigned_facility"] = None
        elif validated_data.get("assigned_facility"):
            validated_data["assigned_facility_external"] = None
            assigned_facility_external_id = validated_data.pop("assigned_facility")[
                "external_id"
            ]
            if assigned_facility_external_id:

                validated_data["assigned_facility_id"] = Facility.objects.get(
                    external_id=assigned_facility_external_id
                ).id
```

Should I go ahead with this change?